packages/pubsub/src/subscription.js

```diff
@@ -592,30 +350,154 @@ Subscription.prototype.decorateMessage_ = function(message) {
  *   var apiResponse = data[0];
  * });
  */
-Subscription.prototype.delete = function(callback) {
+Subscription.prototype.delete = function(gaxOpts, callback) {
   var self = this;
 
-  callback = callback || common.util.noop;
+  if (is.fn(gaxOpts)) {
+    callback = gaxOpts;
+    gaxOpts = {};
+  }
 
-  var protoOpts = {
-    service: 'Subscriber',
-    method: 'deleteSubscription'
+  var reqOpts = {
+    subscription: this.name
   };
 
+  this.request({
+    client: 'subscriberClient',
+    method: 'deleteSubscription',
+    reqOpts: reqOpts,
+    gaxOpts: gaxOpts
+  }, function(err, resp) {
+    if (!err) {
+      self.closeConnection_();
```

Comment:
I never suggested removing error listeners. I don't recall ever mentioning error listeners.

Anyway -- I'm trying to come up with our approach to error handling should be, and then make the code reflect that. It seems like we have 3 modes:

1. As long as there are zero listeners for the message event, we want the connection to be closed
2. As long there are non-zero listeners for the message event, we want the connection to be open (so we retry opening the connection indefinitely. And once we establish a connection, if it goes down, we must start retrying to re-establish again)...
3. Until we decide to permanently give up. We need a very good reason for this (we got a subscription not found error, or a topic not found error, or a permission denied error, or we just retried opening the connection for some significant period of time and could not)

Do we agree on this? If so, then there's a couple of important points:

1. While we are in mode #3, then we should not call openConnection (this is not true of the code right now. openConnection is called any time a message event listener is added)
2. If a new message event listener is added while we are in mode #3, we should probably emit an error

Does this make sense?